### PR TITLE
RCAL-806: FY24 Datamodels Document Review

### LINKED ISSUE #386: FY24 Document Review
Review RDM documentation to re-align any generated discontinuities.

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RCAL-806](https://jira.stsci.edu/browse/RCAL-806)

Closes #386 #341 

This PR updates the documentation to match the present code version.

## Tasks
- [ ] Update or add relevant `roman_datamodels` tests.
- [x] Update relevant docstrings and / or `docs/` page.
- [ ] Does this PR change any API used downstream? (If not, label with `no-changelog-entry-needed`.)
  - [ ] Write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types).
  - [ ] Start a `romancal` regression test (https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/rad@<branch>"`).

<details><summary>News fragment change types:</summary>

- ``changes/<PR#>.feature.rst``: new feature
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.doc.rst``: documentation change
- ``changes/<PR#>.removal.rst``: deprecation or removal of public API
- ``changes/<PR#>.misc.rst``: infrastructure or miscellaneous change
</details

### CODE REVIEW

**Review by schlafly — COMMENTED** [review]
This looks good to me.  IIRC Brett just removed assignment validation from roman_datamodels so I don't think we should document that bit.

File: docs/roman_datamodels/datamodels/using_datamodels.rst [hunk 1]
```diff
@@ -19,49 +19,54 @@ page::
     >>> dm = rdm.open('my_wonderful_roman_data.asdf')
 
     >>> type(dm)
-    roman_datamodels.datamodels.ScienceRawModel
+    <class 'roman_datamodels.datamodels._datamodels.ScienceRawModel'>
+
+    # Can view the associated schema
+    >>> dm.tag
+    'asdf://stsci.edu/datamodels/roman/tags/wfi_science_raw-1.0.0'
```
Comment (braingram) [inline docs/roman_datamodels/datamodels/using_datamodels.rst:26]:
```suggestion
    >>> dm.schema_uri
    'asdf://stsci.edu/datamodels/roman/schemas/wfi_science_raw-1.0.0'
```
Or alternatively change "schema" on line 24 to "tag".

File: docs/roman_datamodels/datamodels/using_datamodels.rst [hunk 2]
```diff
@@ -19,49 +19,54 @@ page::
...
-    >>> dm.meta.exposure.start_time_mjd = 60000.
-    print(dm.meta.exposure.start_time_mjd)
+    >>> dm.meta.exposure.exposure_time = 60000.
+    >>> print(dm.meta.exposure.exposure_time)
     60000.0
 
     # Try to assign invalid type
 
-    >>> d.meta.exposure.start_time_mjd = "hello"
+    >>> dm.meta.exposure.exposure_time = "hello"
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply docs/roman_datamodels/datamodels/using_datamodels.rst:50]: Added to this PR.

File: docs/roman_datamodels/datamodels/using_datamodels.rst (hunk 1 above)
Comment (PaulHuwe) [reply docs/roman_datamodels/datamodels/using_datamodels.rst:26]: Changed.

**Review by braingram — APPROVED** [review]
LGTM. Thanks for all the fixes.

**Pull request merged**